front/src/modules/simulationResult/components/SpaceTimeChartWrapper/useTrackOccupancy.ts

```diff
@@ -162,35 +196,136 @@ const useTrackOccupancy = ({
-                  train_schedule_ids: Object.keys(trainsCollection)
-                    .filter((id) => isTrainScheduleId(id))
-                    .map((id) => extractEditoastIdFromTrainScheduleId(id)),
-                },
-              })
-            ).data,
-            (entries, trackId) =>
-              entries.map((entry) =>
-                getMovableOccupancyZone(
-                  trackId,
-                  entry,
-                  trainsCollection[formatEditoastIdToTrainScheduleId(entry.train_schedule_id)]
-                )
+    ): Promise<MovableOccupancyZone[]> => {
+      if (!opId) return [];
+
+      const scheduleIds: TrainScheduleId[] = [];
+      const pacedIds: PacedTrainId[] = [];
+      for (const id of Object.keys(trainsCollection)) {
+        if (isTrainScheduleId(id)) scheduleIds.push(id);
+        else if (isPacedTrainId(id)) pacedIds.push(id);
+      }
+
+      const bodyForTrainSchedules =
+        scheduleIds.length > 0
+          ? {
+              operational_point_id: opId,
+              infra_id: infraId,
+              train_schedule_ids: scheduleIds.map(extractEditoastIdFromTrainScheduleId),
+            }
+          : null;
+
+      const bodyForPaced =
+        pacedIds.length > 0
+          ? {
+              operational_point_id: opId,
+              infra_id: infraId,
+              paced_train_ids: pacedIds.map(extractEditoastIdFromPacedTrainId),
+            }
+          : null;
+
+      const [trainScheduleResp, pacedResp] = await Promise.all([
+        bodyForTrainSchedules
+          ? postTrainScheduleTrackOccupancy({ body: bodyForTrainSchedules })
+          : Promise.resolve(undefined),
+        bodyForPaced
+          ? postPacedTrainTrackOccupancy({ body: bodyForPaced })
+          : Promise.resolve(undefined),
+      ]);
+
+      const zones: MovableOccupancyZone[] = [];
+
+      if (trainScheduleResp?.data) {
+        for (const [trackId, occupations] of Object.entries(trainScheduleResp.data)) {
+          for (const occupation of occupations) {
+            const trainId = formatEditoastIdToTrainScheduleId(occupation.train_schedule_id);
+            const train = trainsCollection[trainId];
+            if (!train) continue;
+
+            zones.push(
+              getMovableOccupancyZone(
+                trackId,
+                trainId,
+                occupation,
+                train.spaceTimeCurves,
+                train.name,
+                train.departureTime
+              )
+            );
+          }
+        }
+      }
+
+      if (pacedResp?.data) {
+        for (const [trackId, occupations] of Object.entries(pacedResp.data)) {
+          for (const occupation of occupations) {
+            const pacedId = formatEditoastIdToPacedTrainId(occupation.paced_train_id);
+            const train = trainsCollection[pacedId];
+            if (!train) continue;
+
+            if (!('paced' in train)) {
+              continue;
+            }
```

Comment:
You can use the function `isTrainScheduleProjection`
```suggestion
            if (!train || isTrainScheduleProjection(train)) {
              continue;
            }
```